### PR TITLE
Add keep file path for ios

### LINKED ISSUE #705: Add keep-file-path for iOS
**Summary**
We use flank for our iOS project and we're trying to capture all `.xcresult` bundles generated from a test run. When we attempt to download these, they show up in our results directory with a flattened structure. Opening this issue here to see if we can add functionality to download these bundles without flattening their contents.

**Details**
We've tried three different approaches:

**1**
```
  files-to-download:
    - .*\.xcresult.*$
```
**2**
```
  files-to-download:
    - .*\.xcresult.*$
  keep-file-path: true
```
**3** 
```
  directories-to-download:
    - .*\.xcresult.*$
```

The last one did not work. The first two successfully copied the contents we need, but in a flattened directory structure. So, instead of something like this:

```
├── shard_0
│   └── iphonexr-13.2-en-portrait
│       ├── TestLogs
│       │   └── Test-Transient\ Testing-2020.04.07_17-23-56--0700.xcresult
│       │       ├── Data
│       │       │   ├── data.0~0PgiStZxX4SUtgntTpMiIGcJCvOs-CE4HIfAo7hOUDcOQyIgR4x5QBRFRNTzcod2egVED0LT1AIAwSR2YWuBeg==
│       │       │   ├── data.0~1wgyNqjDm_8AvJiPn6KK-h-gDFy6I4rSRvSgX6P_zHIH_Gk7hcqw9Qx5vi7e3KfKWHJmheQ7kPj72GVOuMLr0g==
│       │       │   ├── data.0~4IDFLGJhx3gAaPyanxieAsFhugrVaRhRtxggw4sGIFxExib3-fP3OY9FlM70eZ69f-FfehBHMWkXA43mroYmjw==
│       │       │   ├── refs.0~Y5AfVAwFExe9DkMflLoAbTbG9CyJ-koYWk3xRBa7qDy3wWndIiJUKr-nvzjl6auKUnwfUHRsWmxq99OCeYiHEw==
│       │       │   ├── refs.0~yRzPrr6sOdDqbaBBFVcYHoSECPqtpzXxiGGydmEv5_YvNBCnJnCpCICDTd_q230WApjHv0EC-qG_BzXpPC87Rg==
│       │       │   └── refs.0~zZtlolMc76PkmfiriEpadxGOj1kzo2OB7j-GOc7ljL-wrY0rqu5DtraUrs0Xnxw4yobUiDcF7YHBHW55Wbq_vA==
│       │       └── info.plist
│       └── test_result_0.xml
└── shard_1
    └── iphonexr-13.2-en-portrait
        ├── TestLogs
        │   └── Test-Transient\ Testing-2020.04.07_17-24-43--0700.xcresult
        │       ├── Data
        │       │   ├── data.0~01b2Baj03tScx1v_yiWwbs66oWipqyP-AKW8y_UhuTqMRQa1yNgwFiIms21bfnGcHEb12_2nl20ScR6aedbdIA==
        │       │   ├── data.0~0ceqCl8g4ANq20e1I0yvhLETlpjm5D0CH_A60gttIVWotxUf62d_i1j4NKl2ug84AyPdaqVBuzPS5A6l7LG53A==
        │       │   ├── data.0~2gnk-5mBAXc8bLRIldjdp8gVe7TTkcw3HjIJJJhMeIZNiwro80snAUdu1rLNW3EjA5hbknwD7n2rWG6le-qObw==
        │       │   ├── refs.0~xkTrSrsvMAKQhQJsJY4T0q61sVkmvqx7z2nu_MKhvPTy3HZWx8CMPL40Q4OhPxIDjZH1dgkcw7Ep3XSI0iMyyQ==
        │       │   ├── refs.0~z3y7rvyK0zfJgppu9raJp127gASTlZDxhMfMzL_hTrsxu2aqPNKqKOBdqh9M-YdU_aMbIwS0cUFUYYwqpiSYyw==
        │       │   └── refs.0~zOhUVC00ejw8BrkTYjBZYLEp9ur5PLJ_6n8JS9TzyUNC3vt3yCx3gCfsHHGSlCDJamErm8zMDpwyeh263k7C3A==
        │       └── info.plist
        └── test_result_0.xml
```

We get this:
```
├── shard_0
│   └── iphonexr-13.2-en-portrait
│       ├── Info.plist
│       ├── data.0~01b2Baj03tScx1v_yiWwbs66oWipqyP-AKW8y_UhuTqMRQa1yNgwFiIms21bfnGcHEb12_2nl20ScR6aedbdIA==
│       ├── data.0~0ceqCl8g4ANq20e1I0yvhLETlpjm5D0CH_A60gttIVWotxUf62d_i1j4NKl2ug84AyPdaqVBuzPS5A6l7LG53A==
│       ├── data.0~2gnk-5mBAXc8bLRIldjdp8gVe7TTkcw3HjIJJJhMeIZNiwro80snAUdu1rLNW3EjA5hbknwD7n2rWG6le-qObw==
│       ├── refs.0~xkTrSrsvMAKQhQJsJY4T0q61sVkmvqx7z2nu_MKhvPTy3HZWx8CMPL40Q4OhPxIDjZH1dgkcw7Ep3XSI0iMyyQ==
│       ├── refs.0~z3y7rvyK0zfJgppu9raJp127gASTlZDxhMfMzL_hTrsxu2aqPNKqKOBdqh9M-YdU_aMbIwS0cUFUYYwqpiSYyw==
│       ├── refs.0~zOhUVC00ejw8BrkTYjBZYLEp9ur5PLJ_6n8JS9TzyUNC3vt3yCx3gCfsHHGSlCDJamErm8zMDpwyeh263k7C3A==
│       └── test_result_0.xml
└── shard_1
    └── iphonexr-13.2-en-portrait
        ├── Info.plist
        ├── data.0~2-n_7hJz6l0ia9tc0ApeGnlU6AeQ7xwau3mBn3WHvPNf5JsKBER5FqbkHeepZHXSd2eEfRL2Q4P1_SqcjAzG2w==
        ├── data.0~4zr6Cz9KF55fTYwVg9xGGTXmvPhixUaDWGeqboEloeb3wI_EpgeSo6GLPg_3GoNGCA3OSRcjgvoqf9TNlHdZuA==
        ├── data.0~_zgk6csVhhz_P8QIyXTZ5TZcOuWqwV__saP-K6lJj4tSuqee8vWmCJu0L2UmzEnOnwUbGwOIbaNouaSJtYMYAw==
        ├── refs.0~XCuprdkG2bzRKdCfPT3Tsy2OZiiC7l5D2Jb3W3ElyaUqb1cJWVeHAfc2RDOuouKWTcbC85udql6NHp0SWR2pdw==
        ├── refs.0~XIOUbGFVjhcyQcXTO7i1rWI-hXLywlNdloiYIftDf_ehRTd3LY9JPq_X-yaXaRDsdMEiEvUeCYwTwrNYpHrcBg==
        ├── refs.0~w70ytisQ6PC1mH6apQkmk9CeySLL43DZKGdaSfIAsKE7lQFB83GzdFyAfo4ugK7kB-MVqgz4EeuEqgXILtss_g==
        └── test_result_0.xml
```

We can post-process the files and restore their structure, but it would be nice to have Flank do this for us instead. Thanks in advance for considering this.

### PR DESCRIPTION
**Pull request by pawelpasterz**
Fixes #705 

## Checklist

- [x] Documented
- [x] Unit tested
- [x] release_notes.md updated

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -168,6 +168,10 @@ flank:
   ## Local folder to store the test result. Folder is DELETED before each run to ensure only artifacts from the new run are saved.
   # local-result-dir: flank
 
+  ## Keeps the full path of downloaded files. Required when file names are not unique.
+  ## Default: false  
+  # keep-file-path: false
```
Comment (bootstraponline) [inline README.md:173]: Make sure to update `flank.ios.yml`
https://github.com/Flank/flank/blob/master/test_runner/flank.ios.yml

**Comment by bootstraponline** [comment]
Thank you for working on this feature! I've downloaded the branch and ran 3 tests. The feature appears to be working as expected. 🎉 

- Default `flank.ios.yml`, both  `files-to-download` and `keep-file-path` commented out




- Modified `flank.ios.yml`. `files-to-download` enabled. `keep-file-path` set to false

```yaml
 files-to-download:
    - .*
  keep-file-path: false
```




- Modified `flank.ios.yml`. `files-to-download` enabled. `keep-file-path` set to true

```yaml
 files-to-download:
    - .*
  keep-file-path: true
```

File: README.md (hunk 1 above)
Comment (pawelpasterz) [reply README.md:173]: Thanks for pointing this out!

File: test_runner/flank.ios.yml [hunk 2]
```diff
@@ -126,3 +126,7 @@ flank:
 
   ## The max time this test run can execute before it is cancelled (default: unlimited).
   # run-timeout: 60m
+
+  ## Keeps the full path of downloaded files. Required when file names are not unique.
+  ## Default: false
+  # keep-file-path: false
```
Comment (bootstraponline) [inline test_runner/flank.ios.yml:132]: image

let's keep the newline at the end of the file
https://unix.stackexchange.com/a/18789

**Pull request merged**